custom/icds_reports/management/commands/data_pull_private.py

```diff
@@ -0,0 +1,69 @@
+import csv
+import datetime
+
+from dateutil import parser
+from dateutil.relativedelta import relativedelta
+from django.core.management.base import BaseCommand
+from django.db import connections
+
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from custom.icds_reports.models import AwcLocation
+from custom.icds_reports.utils.connections import get_icds_ucr_citus_db_alias
+
+db_alias = get_icds_ucr_citus_db_alias()
+
+STATE_ID = '3518687a1a6e4b299dedfef967f29c0c'
+
+
+def _run_custom_sql_script(command):
+    with connections[db_alias].cursor() as cursor:
+        cursor.execute(command)
+        return cursor.fetchall()
+
+
+class Command(BaseCommand):
+
+    def get_district_ids_dict(self):
+        filters = {'state_id': STATE_ID, 'aggregation_level': 2}
+        locations = AwcLocation.objects.filter(**filters).values('district_id', 'district_name')
+        district_names = {}
+        for loc in locations:
+            district_names[loc['district_id']] = loc['district_name']
+        return district_names
+
+    def handle(self, **options):
+        query = """
+            SELECT "awc_location_months"."district_id", "child_health_monthly"."child_person_case_id" FROM "public"."child_health_monthly" "child_health_monthly"
+            LEFT JOIN "public"."awc_location_months" "awc_location_months" ON (
+                ("awc_location_months"."month" = "child_health_monthly"."month") AND
+                ("awc_location_months"."awc_id" = "child_health_monthly"."awc_id")
+            ) WHERE "child_health_monthly".month='{month}' AND "awc_location_months".state_id='{state_id}' AND "child_health_monthly".pse_eligible=1;
+        """
+
+        months = [datetime.date(2020, 7, 1), datetime.date(2020, 8, 1)]
+        case_accessor = CaseAccessors('icds-cas')
+        get_district_ids = self.get_district_ids_dict()
+        for month in months:
+            excel_data = [['district', 'count']]
+            person_case_ids = {}
+            for district_id in get_district_ids.keys():
+                person_case_ids[district_id] = []
```

Comment:
can you use the [defaultdict](https://www.geeksforgeeks.org/defaultdict-in-python/) here?

Reply:
updated!